This PR reintroduces an "off-by-one" error in `test_pickle`:

```
======================================================================
FAIL: test_unpickler (test.test_pickle.SizeofTests.test_unpickler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/home/jccl4/cpython/Lib/test/test_pickle.py", line 355, in test_unpickler
    self.assertEqual(object.__sizeof__(u), basesize)
AssertionError: 480 != 472

----------------------------------------------------------------------
```

--- a/Include/pointerobject.h
+++ b/Include/pointerobject.h
@@ -17,6 +17,7 @@ PyAPI_DATA(PyTypeObject) _PyNativePointer_Type;
 PyAPI_FUNC(_Bool) PyNativePointer_IsNull(PyObject *);
 PyAPI_FUNC(PyObject *) PyNativePointer_FromVoidPointer(void *);
 PyAPI_FUNC(void *) PyNativePointer_AsVoidPointer(PyObject *);
+PyAPI_FUNC(PyObject *) PyNativePointer_FromUIntPtr(uintptr_t);
 PyAPI_FUNC(uintptr_t) PyNativePointer_AsUIntPtr(PyObject *);
 
 #ifdef __cplusplus

--- a/Lib/test/support/__init__.py
+++ b/Lib/test/support/__init__.py
@@ -809,8 +809,6 @@ _align = '0n'
 if hasattr(sys, "getobjects"):
     _header = '2P' + _header
     _align = '0P'
-if hasattr(sys, "cheri_platform"):
-    _align = '0P'
 _vheader = _header + 'n'
 
 def calcobjsize(fmt):

--- a/Objects/memoryobject.c
+++ b/Objects/memoryobject.c
@@ -1832,7 +1832,7 @@ convert_bool:
 convert_bytes:
     return PyBytes_FromStringAndSize(ptr, 1);
 convert_pointer:
-    return PyNativePointer_FromVoidPointer(p);
+    return PyNativePointer_FromUIntPtr((uintptr_t)p);
 err_format:
     PyErr_Format(PyExc_NotImplementedError,
         "memoryview: format %s not supported", fmt);

--- a/Objects/pointerobject.c
+++ b/Objects/pointerobject.c
@@ -219,6 +219,23 @@ PyNativePointer_AsVoidPointer(PyObject *vv)
 #endif
 }
 
+PyObject *
+PyNativePointer_FromUIntPtr(uintptr_t value) {
+    /* A parallel to PyNativePointer_FromVoidPointer method, allowing */
+    /* untagged uintptr_t, with the developer assuming the risks of this. */
+    PyNativePointerObject *p;
+    if (value == NULL) {
+        Py_INCREF(Py_NativeNULL);
+        return Py_NativeNULL;
+    }
+    /* Cheri validity tag check is skipped here, in constrast to FromVoidPointer method. */
+    p = PyObject_New(PyNativePointerObject, &_PyNativePointer_Type);
+    if (p == NULL)
+        return NULL;
+    p->pointer = (void *)value;
+    return (PyObject *)p;
+}
+
 uintptr_t
 PyNativePointer_AsUIntPtr(PyObject *vv)
 {

